src/components/SearchBox.js

```diff
@@ -228,45 +253,148 @@ class SearchBox extends Component {
     }
   };
 
+  renderIcons = () => {
+    const {
+      iconPosition,
+      showClear,
+      clearIcon,
+      getMicInstance,
+      renderMic,
+      innerClass,
+      showVoiceSearch,
+      icon,
+      showIcon
+    } = this.props;
+    const { currentValue, micStatus } = this.state;
+    return (
+      <Icons
+        clearValue={this.clearValue}
+        iconPosition={iconPosition}
+        showClear={showClear}
+        clearIcon={clearIcon}
+        currentValue={currentValue}
+        handleSearchIconClick={this.handleSearchIconClick}
+        icon={icon}
+        showIcon={showIcon}
+        getMicInstance={getMicInstance}
+        renderMic={renderMic}
+        innerClass={innerClass}
+        enableVoiceSearch={showVoiceSearch}
+        onMicClick={() => {
+          this.searchBase &&
+            this.searchBase.onMicClick(null, { triggerSuggestionsQuery: true });
+        }}
+        micStatus={micStatus}
+      />
+    );
+  };
+
+  renderNoSuggestion = () => {
+    const { renderNoSuggestion, innerClass, renderError } = this.props;
+    const {
+      loading,
+      error,
+      isOpen,
+      currentValue,
+      suggestionsList
+    } = this.state;
+    if (
+      renderNoSuggestion &&
+      isOpen &&
+      !suggestionsList.length &&
+      !loading &&
+      currentValue &&
+      !(renderError && error)
+    ) {
+      return (
+        <SuggestionWrapper
+          className="no-suggestions"
+          innerClass={innerClass}
+          innerClassName="noSuggestion"
+        >
+          {typeof renderNoSuggestion === 'function'
+            ? renderNoSuggestion(currentValue)
+            : renderNoSuggestion}
+        </SuggestionWrapper>
+      );
+    }
+    return null;
+  };
+
+  renderError = () => {
```

Comment:
@ShahAnuj2610 do we have `renderError` in Reactivesearch? If not we can remove this.

Reply:
yes, it's there

Reply:
Can you point the line please I can check how its being used.

Reply:
https://github.com/ShahAnuj2610/react-searchbox/blob/f6873552f45584e79548f352d2aa47f5d203140b/src/components/SearchBox.js#L445

Reply:
Not in Searchbox in Reactivesearch.

Reply:
RS: https://github.com/appbaseio/reactivesearch/blob/c2b18ed9e176f1d87c12db883f202c382f7f67d9/packages/web/src/components/search/DataSearch.js#L846

Reply:
Okay cool 👍